From 87b7fec6195bab03daeff2cb4b526e8472307bce Mon Sep 17 00:00:00 2001
From: Damien Le Moal <damien.lemoal@wdc.com>
Date: Thu, 19 Aug 2021 16:19:44 +0900
Subject: [PATCH 22/31] libata: improve ata_scsiop_maint_in()

Allow translation of the REPORT_SUPPORTED_OPERATION_CODES commands
using the command format 0x3, that is, REPORT_SUPPORTED_OPERATION_CODES
commands checking support for commands identified using an opcode and
a service action.

Signed-off-by: Damien Le Moal <damien.lemoal@wdc.com>
---
 drivers/ata/libata-scsi.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/ata/libata-scsi.c b/drivers/ata/libata-scsi.c
index 1336bf6bd9ea..bc9634237fc6 100644
--- a/drivers/ata/libata-scsi.c
+++ b/drivers/ata/libata-scsi.c
@@ -3236,11 +3236,12 @@ static unsigned int ata_scsiop_maint_in(struct ata_scsi_args *args, u8 *rbuf)
 	u8 supported = 0;
 	unsigned int err = 0;
 
-	if (cdb[2] != 1) {
+	if (cdb[2] != 1 && cdb[2] != 3) {
 		ata_dev_warn(dev, "invalid command format %d\n", cdb[2]);
 		err = 2;
 		goto out;
 	}
+
 	switch (cdb[3]) {
 	case INQUIRY:
 	case MODE_SENSE:
-- 
2.31.1

